### PR TITLE
Flow type rule nodes: added tests

### PR DESCRIPTION
**Pull request by irynamatveieva**
## Pull Request description

Added tests for the following flow type rule nodes:
- acknowledge 
- checkpoint
- output

There are no conflicts with PE.

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java [hunk 1]
```diff
@@ -16,17 +16,88 @@
...
+        assertThat(config.getVersion()).isEqualTo(0);
+    }
+
+    @Test
+    public void givenDefaultConfig_whenInit_thenOk() {
+        assertThatNoException().isThrownBy(() -> node.init(ctxMock, nodeConfiguration));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {DataConstants.MAIN_QUEUE_NAME, DataConstants.HP_QUEUE_NAME, DataConstants.HP_QUEUE_NAME, "Custom queue"})
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java:83]:
```suggestion
    @ValueSource(strings = {DataConstants.MAIN_QUEUE_NAME, DataConstants.HP_QUEUE_NAME, DataConstants.SQ_QUEUE_NAME, "Custom queue"})
```

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbAckNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.mockito.BDDMockito.then;
+
+@ExtendWith(MockitoExtension.class)
+public class TbAckNodeTest {
+
+    private TbAckNode node;
+    private EmptyNodeConfiguration config;
+    private TbNodeConfiguration nodeConfiguration;
+
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbAckNodeTest.java:45]: please remove empty line

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java [hunk 3]
```diff
@@ -16,17 +16,88 @@
...
+    public void givenQueueName_whenOnMsg_thenTransfersMsgToDefinedQueue(String queueName) throws TbNodeException {
+        given(ctxMock.getQueueName()).willReturn(queueName);
+        willAnswer(invocationOnMock -> {
+            Runnable onSuccess = invocationOnMock.getArgument(3);
+            onSuccess.run();
+            return null;
+        }).given(ctxMock).enqueueForTellNext(any(TbMsg.class), any(String.class), any(String.class), any(Runnable.class), any(Consumer.class));
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java:90]: Please remove this and use ArgumentCaptor in the enqueueForTellNext verify.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java [hunk 4]
```diff
@@ -16,17 +16,88 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.spy;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.BDDMockito.willAnswer;
 
 @Slf4j
+@ExtendWith(MockitoExtension.class)
 public class TbCheckpointNodeTest extends AbstractRuleNodeUpgradeTest {
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/flow/TbCheckpointNodeTest.java:56]: Please add a negative test case for the onMsg method

**Review by ShvaykaD — CHANGES_REQUESTED** [review]

**Review by ShvaykaD — APPROVED** [review]

**Review by ViacheslavKlimov — APPROVED** [review]

**Pull request merged**